larray/inout/csv.py

```diff
@@ -70,58 +70,133 @@ def read_csv(filepath_or_buffer, nb_axes=None, index_col=None, sep=',', headerse
 
     Examples
     --------
-    >>> tmpdir = getfixture('tmpdir')
-    >>> fname = os.path.join(tmpdir.strpath, 'test.csv')
-    >>> a = ndtest('nat=BE,FO;sex=M,F')
-    >>> a
-    nat\\sex  M  F
-         BE  0  1
-         FO  2  3
-    >>> a.to_csv(fname)
+    >>> import os
+    >>> from larray import EXAMPLE_FILES_DIR
+    >>> fname = os.path.join(EXAMPLE_FILES_DIR, 'test3d.csv')
+    >>> read_csv(fname)
+    a  b\c  c0  c1  c2
+    1   b0   0   1   2
+    1   b1   3   4   5
+    2   b0   6   7   8
+    2   b1   9  10  11
+    3   b0  12  13  14
+    3   b1  15  16  17
+
+    Missing label combinations
+
+    >>> fname = os.path.join(EXAMPLE_FILES_DIR, 'testmissing_values.csv')
+    >>> # let's take a look inside the CSV file.
+    >>> # they are missing label combinations: (a=2, b=b0) and (a=3, b=b1)
     >>> with open(fname) as f:
     ...     print(f.read().strip())
-    nat\\sex,M,F
-    BE,0,1
-    FO,2,3
+    a,b\c,c0,c1,c2
+    1,b0,0,1,2
+    1,b1,3,4,5
+    2,b1,9,10,11
+    3,b0,12,13,14
+    >>> # by default, cells associated with missing label combinations are fulfilled with NaN.
+    >>> # In that case, an int array are converted to a float array.
     >>> read_csv(fname)
-    nat\\sex  M  F
-         BE  0  1
-         FO  2  3
-
-    Sort columns
+    a  b\c    c0    c1    c2
+    1   b0   0.0   1.0   2.0
+    1   b1   3.0   4.0   5.0
+    2   b0   nan   nan   nan
+    2   b1   9.0  10.0  11.0
+    3   b0  12.0  13.0  14.0
+    3   b1   nan   nan   nan
+    >>> # using argument 'fill_value', you can choose which value to set in cells
+    >>> # associated with missing label .
+    >>> read_csv(fname, fill_value=0)
+    a  b\c  c0  c1  c2
+    1   b0   0   1   2
+    1   b1   3   4   5
+    2   b0   0   0   0
+    2   b1   9  10  11
+    3   b0  12  13  14
+    3   b1   0   0   0
 
-    >>> read_csv(fname, sort_columns=True)
-    nat\\sex  F  M
-         BE  1  0
-         FO  3  2
-
-    Read array saved in "narrow" format (wide=False)
+    Specify the number of axes of the output array (useful when the name of the last axis is implicit)
 
-    >>> a.to_csv(fname, wide=False)
+    >>> fname = os.path.join(EXAMPLE_FILES_DIR, 'test2d_classic.csv')
+    >>> # let's take a look inside the CSV file.
+    >>> # The name of the second axis is missing.
     >>> with open(fname) as f:
     ...     print(f.read().strip())
-    nat,sex,value
-    BE,M,0
-    BE,F,1
-    FO,M,2
-    FO,F,3
-    >>> read_csv(fname, wide=False)
-    nat\\sex  M  F
-         BE  0  1
-         FO  2  3
+    a,b0,b1,b2
+    a0,0,1,2
+    a1,3,4,5
+    a2,6,7,8
+    >>> # read the array stored in the CSV file as it
+    >>> read_csv(fname)
```

Comment:
as I told you the other day, I think we should break this case. ie make it return:
```
{0}   a  b0  b1  b2
  0  a0   0   1   2
  1  a1   3   4   5
  2  a2   6   7   8
```
because there is no guarantee the first column is an index